File: src/main/java/com/christensen/prsapp/repositories/entities/Player.java
package com.christensen.prsapp.repositories.entities;

import com.christensen.prsapp.controllers.dto.ResultEnum;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.Id;

@Data
@Entity
public class Player {

    @Id
    private String name;
    private int wins;
    private int draws;
    private int losses;

    public void setResult(ResultEnum result) {
        switch (result) {
            case WIN:
                wins += 1;
                break;
            case DRAW:
                draws += 1;
                break;
            case LOSS:
                losses += 1;
                break;
        }
    }

    @Override
    public String toString() {
        return "User{" + " name=" + name + ", wins=" + wins + ", draws=" + draws + ", losses=" + losses + '}';
    }
}

File: angularclient/src/app/model/player.ts
import { Injectable } from "@angular/core";

@Injectable()
export class Player {
    name: string;
    wins: number;
    draws: number;
    losses: number;

    map(player: Player) {
      this.name = player.name;
      this.wins = player.wins;
      this.draws = player.draws;
      this.losses = player.losses;
    }
}
